src/api/routes/tx.ts

```diff
@@ -250,6 +254,26 @@ export function createTxRouter(db: DataStore): express.Router {
     })
   );
 
+  router.get(
+    '/events',
+    asyncHandler(async (req, res, next) => {
+      const limit = parseTxQueryEventsLimit(req.query['limit'] ?? 96);
+      const offset = parsePagingQueryInput(req.query['offset'] ?? 0);
+
+      const principalOrTxId = parseAddressOrTxId(req, res, next);
+      const eventTypeFilter = parseEventTypeFilter(req, res, next);
+
+      const { results } = await db.getTransactionEvents({
+        addressOrTxId: principalOrTxId,
+        eventTypeFilter,
+        offset,
+        limit,
+      });
+      const response = { limit, offset, events: results.map(e => parseDbEvent(e)) };
+      res.status(200).json(response);
```

Comment:
Can you add the chain tip cache headers to this response? You can look at the `/block` endpoint for an example on how to do that